index.js

```diff
@@ -193,53 +193,23 @@ module.exports = class Hyperdrive extends ReadyResource {
   async clear (name, opts) {
     if (!this.opened) await this.ready()
 
-    const diffing = !!(opts && opts.diff)
+    const node = await this.entry(name, { wait: false }).catch(nullifyCatch)
 
-    let node = null
-
-    try {
-      node = await this.entry(name, { wait: false })
-    } catch {
-      // do nothing, prop not available
-    }
-
-    const infoBefore = diffing === true
-      ? await this.blobs.core.info({ storage: true })
-      : null
-
-    if (node !== null && this.blobs !== null) {
-      await this.blobs.clear(node.value.blob)
+    if (node === null || this.blobs === null) {
+      return (opts && opts.diff) ? { blocks: 0 } : null
     }
 
-    if (diffing === false) return null
-
-    const infoAfter = await this.blobs.core.info({ storage: true })
-
-    return {
-      blocks: infoBefore.storage.blocks - infoAfter.storage.blocks
-    }
+    return this.blobs.clear(node.value.blob, opts)
   }
 
   async clearAll (opts) {
     if (!this.opened) await this.ready()
 
-    const diffing = !!(opts && opts.diff)
-
-    const infoBefore = diffing === true
-      ? await this.blobs.core.info({ storage: true })
-      : null
-
-    if (this.blobs !== null) {
-      await this.blobs.core.clear(0, this.blobs.core.length)
+    if (this.blobs === null) {
+      return (opts && opts.diff) ? { blocks: 0 } : null
     }
 
-    if (diffing === false) return null
-
-    const infoAfter = await this.blobs.core.info({ storage: true })
-
-    return {
-      blocks: infoBefore.storage.blocks - infoAfter.storage.blocks
-    }
+    return this.blobs.core.clear(0, this.blobs.core.length, opts)
```

Comment:
Scratch that last remark, it wouldn't solve anything to have a clearAll method in Hyperblobs

Reply:
Common practice everywhere, i.e.:
https://github.com/mafintosh/ready-resource/blob/614535e4d0a65e14b6f8beefd325af24a455fab9/index.js#L20

Or maybe I'm not understanding your concern, `clearAll` is async as it waits for ready at the beginning

Reply:
Ah, my point isn't valid. Because it's declared as an async function, it will always returns a Promise anyway. 
(To clarify: I thought the API of this method was [null|Object|Promise], but I was wrong)